Test Cases (maybe) done

BudiartoSalim opened the pull request:
Model Validations, routing, and controller is done but untested.

Seed is not created yet; will need to generate Seed tomorrow for testing.

Error handler, Authentication and Authorization middlewares not created yet.

Most likely for Update and Delete the response will be changed, some responses are not properly written yet.

BudiartoSalim committed:
Test Cases (maybe) done

Model Validations, routing, and controller is done but untested.

Seed is not created yet; will need to generate Seed tomorrow for testing.

Error handler, Authentication and Authorization middlewares not created yet.

Most likely for Update and Delete the response will be changed, some responses are not properly written yet.

BudiartoSalim committed:
Finally passes all Test cases

BudiartoSalim committed:
Changed so that tests does not require initial seeding

BudiartoSalim committed:
added .env template

BudiartoSalim committed:
added 1 more endpoint to search product by ID just in case, including its test cases. new feature already passes the tests as well.

BudiartoSalim committed:
some changes in validation to match the seed requirements

BudiartoSalim committed:
updated config

BudiartoSalim committed:
fixed package json script

BudiartoSalim committed:
Update readme.md

dhimashary commented:
di app require dotenv jangan bener2 diilangin 100%, tapi dikasih kondisional kalo jalan di development/test require dotenvnya dinyalain

dhimashary commented on middlewares/authentication.js:
```diff
@@ -0,0 +1,16 @@
+const jwt = require('jsonwebtoken');
+
+function authenticateUser(req, res, next) {
```
user findonenya harusnya di auntentication bukan di authorization

BudiartoSalim committed:
improved authentication and authorization. get product by ID no longer requires admin authorization. added ShoppingCart and CartProduct model+migration for future content.

BudiartoSalim committed:
Merge branch 'master' of https://github.com/BudiartoSalim/ecommerce-server

BudiartoSalim committed:
Added Login, Register, Fetch, Upd/del cart item

All features is not tested yet, but Login and Register should work.

Fetch is still very experimental, experimenting on nested includes and extra attributes conditions like excluding for practice/learning purposes.

Update Amount and Delete for CartProduct is also untested but should work theoretically.

Still unsure on the Add To Cart feature; already got some ideas but will decide on tomorrow.

BudiartoSalim committed:
a lot of bugfixes, and some rework on model and controller. should work as intended now.

BudiartoSalim committed:
changed so that fetching product list no longer require authentication

BudiartoSalim committed:
improved the add to cart function to increment instead of replace value in case of dupe

BudiartoSalim committed:
added checkout and purchase history feature, tested

BudiartoSalim committed:
Update readme.md
